meka/srcs/sound/sound.cpp

```diff
@@ -378,12 +392,23 @@ void SoundStream_RenderUpToCurrentTime(t_sound_stream* stream)
     s64 elapsed_cycles = current_cycle - stream->last_rendered_cycle_counter;
 
     //elapsed_cycles = MIN(elapsed_cycles, 10000000);   // FIXME-NEWSOUND: how to handle pause, etc?
-
-    // Convert elapsed cycles in 'frames' unit
-    const int cpu_clock = Sound.CpuClock;
-    const double elapsed_emulated_seconds = (double)((double)elapsed_cycles / (double)cpu_clock);
-
-    const double samples_to_render = stream->samples_leftover + (double)Sound.SampleRate * elapsed_emulated_seconds;
+
+    
+    //The OUTPUT (Wall) sound clock (in Z80cycles/sec) is different from the
+    //emulated sound clock (in Z80cycles/sec) as we may be running at a non-standard FPS (e.g. 10Hz or 200Hz).
+    //We are generating samples for the OUPUT  sound buffers, so we need to base our clock rate and rendered sample
+    //count on the output FPS rate (otherwise we will over/underflow the actual output sound)
+    
+    //We can base the output FPS rate on either fskipper.Throttled_Speed (requested) or fskipper.FPS (measured)
+    //Throttled speed appears to avoid most overflows, but the actually measured FPS rate does better at 10Hz (and might work for unthrottled fps)
+    const double output_FPS = (double) fskipper.Throttled_Speed;
+    const double output_cycles_per_sec = GetZ80CyclesPerFrame(&g_machine) * output_FPS;
+    const double elasped_output_seconds = (double)((double)elapsed_cycles / (double)output_cycles_per_sec);
```

Comment:
A bit keen with the casting here?

Reply:
It's mostly present from the old code. I wasn't sure at the time if this was affecting the sound output or not.